File: app/src/main/java/com/erebor/tomkins/pos/data/model/EncryptedAesField.java
package com.erebor.tomkins.pos.data.model;

import androidx.annotation.NonNull;

import com.erebor.tomkins.pos.factory.SecurityEncryptionFactory;
import com.erebor.tomkins.pos.tools.SecurityEncryption;

public class EncryptedAesField {

    private static final String TAG = "EncryptedAesField";
    private SecurityEncryption encryption;
    private final String encrypted;
    private String decrypted;

    public EncryptedAesField(@NonNull String value) {
        this.encryption = SecurityEncryptionFactory.getInstance().getEncryption();
        this.encrypted = encrypt(value);
    }
    public String getDecryptedValue() {
        try {
            if (decrypted != null) {
                return decrypted;
            }
            return decrypted = encryption.aesDecrypt(encrypted);
        } catch (Exception e) {
            return null;
        }
    }
    public String getEncryptedValue() {
        try {
            return encrypted;
        } catch (Exception e) {
            return null;
        }
    }

    private String encrypt(@NonNull String value) {
        try {
            return encryption.aesEncrypt(value);
        } catch (Exception e) {
            return null;
        }
    }

}

File: app/src/main/java/com/erebor/tomkins/pos/data/model/EncryptedAesField.kt
package com.erebor.tomkins.pos.data.model

import com.erebor.tomkins.pos.factory.SecurityEncryptionFactory
import com.erebor.tomkins.pos.tools.SecurityEncryption

class EncryptedAesField(value: String) {
    private val encryption: SecurityEncryption
    private val encrypted: String
    private var decrypted: String? = null
    val decryptedValue: String?
        get() {
            try {
                if (decrypted != null) {
                    return decrypted
                }
                decrypted = encryption.aesDecrypt(encrypted)
                return decrypted
            } catch (e: Exception) {
                return null
            }

        }
    val encryptedValue: String?
        get() {
            try {
                return encrypted
            } catch (e: Exception) {
                return null
            }

        }

    init {
        this.encryption = SecurityEncryptionFactory.instance.encryption
        this.encrypted = encrypt(value)
    }

    private fun encrypt(value: String): String {
        try {
            return encryption.aesEncrypt(value)
        } catch (e: Exception) {
            return value
        }

    }

    companion object {

        private val TAG = "EncryptedAesField"
    }

}
